x-pack/plugins/data_enhanced/server/search/session/session_service.ts

```diff
@@ -81,6 +83,21 @@ export class BackgroundSessionService implements ISessionService {
     }
   };
 
+  /**
+   * Compiles a KQL Query to fetch sessions by ID.
+   * Done as a performance optimization workaround.
+   */
+  private sessionIdsAsFilters(sessionIds: string[]): KueryNode {
+    return nodeBuilder.or(
+      sessionIds.reduce<KueryNode[]>((filters, id) => {
+        filters.push(
+          nodeBuilder.or([nodeBuilder.is(`${BACKGROUND_SESSION_TYPE}.attributes.sessionId`, id)])
+        );
+        return filters;
+      }, [])
+    );
```

Comment:
I'm pretty sure this is going to be pretty slow (both in Kibana and as an ES query that gets generated), could we do something like this instead?
```suggestion
    return nodeBuilder.or(
      sessionIds.map(id => {
        nodeBuilder.is(`${BACKGROUND_SESSION_TYPE}.attributes.sessionId`, id)
      })
    );
```